flang/test/Lower/intrinsic-procedures.f90

```diff
@@ -213,6 +213,97 @@ subroutine dim_testi(i, j, k)
   k = dim(i, j)
 end subroutine
 
+! DOT_PROD
+! CHECK-LABEL: dot_prod_int_default
+subroutine dot_prod_int_default (x, y, z)
```

Comment:
I suggest adding CHECK-SAME on the arguments. See other intrinsic tests for an example (e.g., reshape)